traffic_ops/app/lib/API/Tenant.pm

```diff
@@ -248,7 +326,17 @@ sub delete {
 	if ( !defined($tenant) ) {
 		return $self->not_found();
 	}	
-	my $name = $self->db->resultset('Tenant')->search( { id => $id } )->get_column('name')->single();
+
+	my $parent_tenant = $tenant->parent_id;		
+	
+	my $tenant_utils = UI::TenantUtils->new($self);
+	my $tenants_data = $tenant_utils->create_tenants_data_from_db(undef);
+	
+	if (!$tenant_utils->is_tenant_resource_accessible($tenants_data, $parent_tenant)) {
```

Comment:
I see the parent as an owner of the child, and as such he is the one to be able to delete it.
It is a matter of taste, but note: anyhow a user must not delete his own tenant...